src/components/AuthForm.tsx

```diff
@@ -7,14 +7,36 @@ import {
   FormControlLabel,
   Checkbox,
 } from "@mui/material";
+import { Link } from "react-router-dom";
+import { useDispatch } from "react-redux";
+import { userLogin } from "redux/actions/userLogin";
 
 const AuthForm: React.FC = () => {
   const [isNew, setIsNew] = React.useState(false);
-
+  const dispatch = useDispatch();
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
     setIsNew(event.target.checked);
   };
 
+  const handleClickLogin = () => {
+    dispatch(userLogin(user));
+  };
+
+  const conditionalProps = isNew
+    ? {} //Sign up props will be here
+    : { onClick: handleClickLogin, component: Link, to: "/" };
+
+  const [user, setUser] = useState({
```

Comment:
Better name - userCreds. Also use type for it here `useState<UserCreds>(...)` as `useState` is generic function.
```
type userCreds = {
  email: string;
  password: string;
};
```